### PR TITLE
LPS-122217 Add button to swap the vocabularies used in Content Audit Graph

### PR DESCRIPTION
**Pull request by darquesdev**
**Description**

As a part of the UX improvements in the Content Audit graph, we want to make it easier to change values represented in each of the axes.

**Screenshot**

Screenshot 2020-09-29 at 10 36 12

### CODE REVIEW

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Review by dgarciasarai — COMMENTED** [review]
Awesome team work! 🚀 

**Two vocabularies selected: Flix axes button is available**

Screenshot 2020-10-19 at 13 08 07

**One vocabulary selected: Flip axes button is disabled**

Screenshot 2020-10-19 at 13 07 44

❤️

**Review by dgarciasarai — APPROVED** [review]

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
ci:test:relevant

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/SwapContentDashboardConfigurationMVCActionCommand.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		throws Exception {
+
+		try {
+			PortletPreferences portletPreferences =
+				actionRequest.getPreferences();
+
+			String[] assetVocabularyNames = portletPreferences.getValues(
+				"assetVocabularyNames", new String[0]);
+
+			if (assetVocabularyNames.length == 2) {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/SwapContentDashboardConfigurationMVCActionCommand.java:56]: Could we use ArrayUtil.reverse here?

**Review by cgoncas — CHANGES_REQUESTED** [review]
I think we should use ArrayUtil.reverse instead of reverse our array manually.

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/SwapContentDashboardConfigurationMVCActionCommand.java (hunk 1 above)
Comment (darquesdev) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/SwapContentDashboardConfigurationMVCActionCommand.java:56]: Of course! good catch @cgoncas!

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:stop

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
I see that the requested changes are already done by @darquesdev 👍  cc: @cgoncas

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java [hunk 2]
```diff
@@ -372,6 +374,15 @@ public Integer getStatus() {
 		return _status;
 	}
 
+	public ActionURL getSwapConfigurationURL() {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java:377]: Could we return a generic PortletURL instead?

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java [hunk 3]
```diff
@@ -382,6 +393,14 @@ public long getUserId() {
 		return _userId;
 	}
 
+	public boolean isSwapConfigurationDisabled() {
```
Comment (cgoncas) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java:396]: Is very hard to read negative boolean condition for humans, better rename it as isSwapConfigurationEnabled

Comment (darquesdev) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java:396]: I could not agree more! thx

File: modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java (hunk 2 above)
Comment (darquesdev) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java:377]: Of course we could, but why would we want to do so? IMHO the method is more explicit like it is, and there's at least an example here `com.liferay.segments.context.vocabulary.internal.display.context.SegmentsContextVocabularyConfigurationDisplayContext#getActionURL`. I'm totally open to change it, just want to be sure I get it correctly

Comment (cgoncas) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/display/context/ContentDashboardAdminDisplayContext.java:377]: That makes sense :)

File: modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/ContentDashboardAdminPortletTest.java [hunk 4]
```diff
@@ -1427,6 +1538,22 @@ private ThemeDisplay _getThemeDisplay(Locale locale) throws Exception {
 		return themeDisplay;
 	}
 
+	private Boolean _isSwapConfigurationDisabled(
```
[the remark this answers is not in the document]
Comment (darquesdev) [reply modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/ContentDashboardAdminPortletTest.java:1541]: I ❤️  this comment, let's keep in mind best practices as much as possible

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**